diff --git a/src/hooks/useGameData.js b/src/hooks/useGameData.js
index 43ecae0..16da96f 100644
--- a/src/hooks/useGameData.js
+++ b/src/hooks/useGameData.js
@@ -8,7 +8,7 @@ const useGameData = (gameId, gameState) => {
     setHand, setIsLoading, setCurrentPlayerId,
     setDeckCount, setCurrentTurn, setTurnOrder,
     setPlayers, setHostId, setWinners, setAsesinoGano,
-    setRoles, setSecretCards,
+    setRoles, setSecretCards, setCanRevealSecrets
   } = gameState;
 
   const hasConnectedRef = useRef(false); // evita reconexiones extras
@@ -82,7 +82,8 @@ const useGameData = (gameId, gameState) => {
         }
       }
     };
-
+    setCanRevealSecrets(false); // Inicialmente no se pueden revelar secretos (esto cambiará cuando se jueguen cartas específicas)
+    
     loadGameData();
   }, [gameId]);
 };
diff --git a/src/hooks/useGameState.js b/src/hooks/useGameState.js
index 8273855..d8daf1b 100644
--- a/src/hooks/useGameState.js
+++ b/src/hooks/useGameState.js
@@ -15,6 +15,10 @@ const useGameState = () => {
   const [hostId, setHostId] = useState(null);
   const [roles, setRoles] = useState({ murdererId: null, accompliceId: null });
   const [secretCards, setSecretCards] = useState([]);
+  const [otherPlayersSecrets, setOtherPlayersSecrets] = useState([]);
+  const [selectedSecretCard, setSelectedSecretCard] = useState(null);
+  const [canRevealSecrets, setCanRevealSecrets] = useState(false); // Esto se seteará cuando se juegue una carta que lo permita
+  const [selectedPlayerForSecrets, setSelectedPlayerForSecrets] = useState(null);
 
   // Estados para el fin de la partida
   const [winners, setWinners] = useState(null);
@@ -53,6 +57,10 @@ const useGameState = () => {
     winners, setWinners,
     asesinoGano, setAsesinoGano,
     secretCards, setSecretCards,
+    otherPlayersSecrets, setOtherPlayersSecrets,
+    selectedSecretCard, setSelectedSecretCard,
+    canRevealSecrets, setCanRevealSecrets,
+    selectedPlayerForSecrets, setSelectedPlayerForSecrets,
     // Derived state
     isMyTurn,
     isDiscardButtonEnabled
diff --git a/src/services/apiService.js b/src/services/apiService.js
index 76675ba..e40978a 100644
--- a/src/services/apiService.js
+++ b/src/services/apiService.js
@@ -107,6 +107,22 @@ const createHttpService = () => {
     });
   }
 
+    const getOtherPlayerSecrets = async (gameId, playerId) => {
+    return request(`/partidas/${gameId}/secretosjugador?id_jugador=${playerId}`, {
+      method: "GET",
+    });
+  }
+
+  const revealSecret = async (gameId, playerId, secretId) => {
+    return request(`/partidas/${gameId}/revelacion`, {
+      method: "PATCH",
+      body: JSON.stringify({
+        id_jugador: playerId,
+        id_unico_secreto: secretId
+      }),
+    });
+  }
+
 
   return {
     createGame,
@@ -122,6 +138,8 @@ const createHttpService = () => {
     drawCards,
     getMySecrets,
     getRoles,
+    getOtherPlayerSecrets,
+    revealSecret,
   };
 };
 
